File: Game.java
import java.util.Scanner;
import java.util.Random;

public class Game {

    private static Scanner scanner = new Scanner(System.in);
    private static Random random = new Random();

    public static void main(String[] args) {

        while (Game.game()) {

            System.out.println("Do you want to play again? (y or n): ");

            String dostop = Game.scanner.nextLine();

            if (dostop.equalsIgnoreCase("n")) {

                break;

            }

        }

        Game.scanner.close();

        System.out.println("Goodbye!");

    }


    private static boolean game() {

        System.out.println("----------Number Guessing Game----------");
        System.out.println("Type 0 at any time to quit!");
        System.out.println("----------------------------------------");

        int mynum = Game.random.nextInt(100) + 1;
        int trys = 0;

        while (true) {
            
            
            System.out.println("Enter a number (1-100): ");
            String innumstr = Game.scanner.nextLine();

            int innum = Integer.parseInt(innumstr);


            if (innum == mynum) {

                System.out.println("You Win!");
                System.out.println(String.format("It took you %d trys!", trys));
                return true;

            } else if (innum == 0) {

                System.out.println(String.format("The number was %d", mynum));
                System.out.println(String.format("You had %d trys!", trys));
                return false;

            } else {

                if (innum > mynum) {

                    System.out.println("Lower!");

                } else if (innum < mynum) {

                    System.out.println("Higher!");

                }

            }

            trys += 1;


        }

    }


}

File: game.py
import random

def game():

    mynum = random.randint(1, 100)
    
    trys = 0
    
    print("----------Number Guessing Game----------")
    print("Type 0 at any time to quit!")
    print("----------------------------------------")
    
    
    while True:
        
        inmum = int(input("Enter a number (1-100): "))
        
        if inmum == mynum:
            
            print("You win!")
            print(f"It took you {trys} trys!")
            return True
        
        elif inmum == 0:
            
            print(f"The number was {mynum}!")
            print(f"You had {trys} trys!")
            return False
            
        else:
            
            if inmum > mynum:
                
                print("Lower!")
                
            elif inmum < mynum:
                
                print("Higher!")
        
        trys += 1
        
        

while game():
    
    dostop = input("Do you want to play again? (y or n): ")
    
    if dostop == "n":
        
        break
        
    
print("Goodbye!")